### DISCUSSION TITLE
Forward planning

### DISCUSSION THREAD

**Discussion by cartbar** [firefly-iii/firefly-iii#4910] [Firefly III]
Hello.

I'm new to Firefly and wanted to know if it supports how I am trying to use it.

At the moment I use Quicken and I have set up numerous scheduled transactions, that cover regular payments like bills and also contribute to savings goals.

At the start of the month, I pay a month's worth of scheduled transactions, so that my accounts show the expected expenditure for the following month.  This tells me how much I need to transfer to my savings account to cover the various savings goals and how much is going to be left in my current account at the end of the month (I've seen some people call it a "safe to spend" amount)

Looking at Firefly, my initial thought was to use Piggy Banks as my Savings Goals and Recurring Transactions as my Scheduled Transactions.  However, having played around with it for a bit, I'm not convinced I'm using it in the correct way.  For one thing, Recurring Transactions only allow a transfer to reference Piggy Banks and you have to have a source and destination account, rather that just allocating money in my savings account to a Piggy Bank.  

It is alson not clear if there is any way to get Firefly to run Recurring Transactions ahead of time,

Does Firefly support what I am trying to do?

Thanks,

Carl

**Comment by cartbar** [comment]
It has occurred to me that I could implement this functionality myself using the API, but I can't see anything in the documentation that supports that.  The Firefly gui appears to use "/piggy-banks/add/X".  If I wanted to add an amount to a Piggy Bank using the API, would that be the right one to use?

Thanks,

Carl

**Reply by JC5** [reply]
https://api-docs.firefly-iii.org/#/piggy_banks

**Reply by cartbar** [reply]
Hi.

Thanks for the reply, but that page gives methods to list, create, view, delete and update Piggy Bank, and to list attachments and events, but there is no method listed to create Piggy Bank events, unless I am mis-understanding the documentation.

GET
​/api​/v1​/piggy_banks
List all piggy banks.
POST
​/api​/v1​/piggy_banks
Store a new piggy bank
GET
​/api​/v1​/piggy_banks​/{id}
Get a single piggy bank.
PUT
​/api​/v1​/piggy_banks​/{id}
Update existing piggy bank.
DELETE
​/api​/v1​/piggy_banks​/{id}
Delete a piggy bank.
GET
​/api​/v1​/piggy_banks​/{id}​/events
List all events linked to a piggy bank.
GET
​/api​/v1​/piggy_banks​/{id}​/attachments
Lists all attachments.

Thanks,

Carl

**Comment by JC5** [comment]
What you want to do is set up budgets for your expected expenses, and bills if you want to. Use piggy banks to save money, not for regular transactions.

**Reply by cartbar** [reply]
Hi.

So how would that work?  Assuming, for example, I want to budget £100 per week on food, and £300 per year on birthday presents (I know, I'm a cheapskate), at the start of the month I want to know how much money I will have left at the end of the month if I put £100 per week in the food budget and £25 per month in the birthday present budget?

Thanks,

Carl

**Reply by JC5** [reply]
You can't. Not like that.
